kernel/common/enumerable.rb

```diff
@@ -351,7 +351,18 @@ def any?
   end
 
   def cycle(many=nil)
-    return to_enum(:cycle, many) unless block_given?
+    unless block_given?
+      size = Proc.new do
```

Comment:
I agree with you! These sizes are not that complex to calculate. There will be others, like the [Array#combination](https://github.com/ruby/ruby/blob/v2_1_5/array.c#L4850) that might be worthwhile calculating it only if needed. Will see when we get there, but in the meantime I'll change these.